dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/JdbcEventStore.java

```diff
@@ -174,16 +183,18 @@
         "lastupdated, " +               // 10
         "attributeoptioncomboid, " +    // 11
         "storedby, " +                  // 12
-        "completedby, " +               // 13
-        "deleted, " +                   // 14
-        "code, " +                      // 15
-        "createdatclient, " +           // 16
-        "lastupdatedatclient, " +       // 17
-        "geometry, " +                  // 18
-        "assigneduserid, " +            // 19
-        "eventdatavalues) " +           // 20
+        "createdbyuserinfo, " +         // 13
+        "lastupdatedbyuserinfo, " +     // 14
+        "completedby, " +               // 15
+        "deleted, " +                   // 16
+        "code, " +                      // 17
+        "createdatclient, " +           // 18
+        "lastupdatedatclient, " +       // 19
+        "geometry, " +                  // 20
+        "assigneduserid, " +            // 21
+        "eventdatavalues) " +           // 22
         // @formatter:on
-        "values ( nextval('programstageinstance_sequence'), ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ? )";
+        "values ( nextval('programstageinstance_sequence'), ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ? )";
 
     private final static String UPDATE_EVENT_SQL = "update programstageinstance set " +
```

Comment:
Could we use named parameters instead of index/? based params. When there is so many variables it's very easy to mix up when it's index based...

Reply:
I agree. I'm going to create a technical ticket to use constants from EventQuery instead of using hard coded column names in the code, so replacing sql parameters with named one could be included in that ticket.